### PR TITLE
Create lunatest-community-picks.lua

### PR DESCRIPTION
**Pull request by outstanding-factorio**
### Brief description of the changes:
---Fill in---

### Tested Changes:
- [X] I've tested the changes locally or with people.
- [ ] I've not tested the changes.

Comprehensive Test Suite for CaptainCommunityPick Module
This PR introduces a robust test suite for the CaptainCommunityPick module, enhancing the reliability and maintainability of our team selection process in Biter Battles.
Key Features:

Comprehensive coverage of basic functionality (pick_order and assign_teams)
Tests for edge cases: single player, large groups, and odd number of players
Specific scenario testing: ties, unanimous votes, and reversed preferences
Error handling checks for empty votes and duplicate votes
Mock Factorio environment setup for isolated testing

Improvements:

Enhanced test flexibility to accommodate potential implementation changes
Balanced assertions to ensure correct functionality without over-constraining the implementation
Clear, descriptive test names for easy understanding and maintenance

This test suite will help maintain the integrity of our team selection process as we continue to develop and refine the Biter Battles scenario.
Reviewers, please pay special attention to:

The coverage of edge cases
The balance between strictness and flexibility in the tests
Any scenarios you feel might be missing based on your experience with the game

### CODE REVIEW

File: tests/lunatest-community-picks.lua [hunk 1]
```diff
@@ -0,0 +1,251 @@
...
+
+function test_pick_order_complete_votes()
+    local community_picks = {
+        player1 = {"player1", "player2", "player3", "player4"},
+        player2 = {"player2", "player1", "player3", "player4"},
+        player3 = {"player3", "player1", "player2", "player4"},
+        player4 = {"player4", "player1", "player2", "player3"}
+    }
+    local result = CaptainCommunityPick.pick_order(community_picks)
+    lunatest.assert_equal(4, #result, "Pick order should contain 4 players")
```
Comment (clifffrey) [inline tests/lunatest-community-picks.lua:124]: I think that this would be clearer if you just did
`assert_equal("player1, player2, player3, player4", table.concat(result, ", "))`
same for a few other test cases probably

File: tests/lunatest-community-picks.lua [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+        player4 = {"player4", "player1", "player2", "player3"}
+    }
+    local result = CaptainCommunityPick.pick_order(community_picks)
+    if result then
+        lunatest.assert_equal(4, #result, "If not nil, pick order should contain 4 players")
+        for _, player in ipairs({"player1", "player2", "player3", "player4"}) do
+            lunatest.assert_not_nil(table_contains(result, player), player .. " should be in the pick order")
+        end
+    end
+    -- Test passes whether result is nil or a valid pick order
```
Comment (clifffrey) [inline tests/lunatest-community-picks.lua:213]: I'd slightly prefer that the test just passes one way (i.e. that the test is just verifying what the current output is)

File: tests/lunatest-community-picks.lua [hunk 3]
```diff
@@ -0,0 +1,251 @@
...
+end
+
+function test_assign_teams_preserves_pick_order()
+    local pick_order = {"player1", "player2", "player3", "player4"}
+    local result = CaptainCommunityPick.assign_teams(pick_order)
+    lunatest.assert_equal(2, #result, "Should have 2 teams")
+    lunatest.assert_equal(2, #result[1], "Team 1 should have 2 players")
+    lunatest.assert_equal(2, #result[2], "Team 2 should have 2 players")
+    -- Check that all players are in a team, but don't assume a specific order
```
Comment (clifffrey) [inline tests/lunatest-community-picks.lua:222]: I think that you could be even stronger here?  something like
```
local teams_as_strings = {}
for _, team in ipairs(result) do
  table.insert(teams_as_strings, table.concat(team, ", "))
end
assert_non_nil(table_contains(teams_as_strings, "player1, player4"))
assert_non_nil(table_contains(teams_as_strings, "player2, player3"))

File: tests/lunatest-community-picks.lua [hunk 4]
```diff
@@ -0,0 +1,251 @@
...
+    end
+    local p1_index = table_contains(result, "player1")
+    local p2_index = table_contains(result, "player2")
+    local p3_index = table_contains(result, "player3")
+    local p4_index = table_contains(result, "player4")
+    lunatest.assert_true(math.abs(p1_index - p2_index) == 1, "player1 and player2 should be adjacent")
+    lunatest.assert_true(math.abs(p3_index - p4_index) == 1, "player3 and player4 should be adjacent")
+end
+
+function test_pick_order_large_group()
```
Comment (clifffrey) [inline tests/lunatest-community-picks.lua:100]: perhaps remove the various `group()` tests right now?  as they are just slightly confusing?

**Comment by clifffrey** [comment]
Closing as there has been no progress in a long time. Feel free to reopen if more progress is made.
